Core/Utilities/ElasticSearchDB.py

```diff
@@ -0,0 +1,279 @@
+from elasticsearch.exceptions   import ConnectionError, TransportError, NotFoundError
+from elasticsearch.helpers      import BulkIndexError
+from elasticsearch              import helpers
+from datetime                   import datetime
+from DIRAC.Core.Utilities       import Time
+
+class ElasticSearchDB( object ):
+  
+  """
+  .. class:: ElasticSearchDB
+
+  :param str url: the url to the database for example: el.cern.ch:9200
+  :param str gDebugFile: is used to save the debug information to a file
+  :param int timeout the default time out to Elasticsearch
+  """
+  __chunk_size = 1000
+  __url = ""
+  __timeout = 120
+  ########################################################################
+  def __init__( self, host, port, debug = False ):
+    """ c'tor
+    :param self: self reference
+    :param str host: name of the database for example: MonitoringDB
+    :param str port: The full name of the database for example: 'Monitoring/MonitoringDB'
+    :param bool debug: save the debug information to a file   
+    """
+    global gDebugFile
+    
+    if not hasattr( self, 'log' ):
+      self.log = gLogger.getSubLogger( 'ElasticSearch' )
+    self.logger = self.log
+    
+    self.__url = "http://%s:%s" % ( host, port )
+        
+    if debug:
+      try:
+        gDebugFile = open( "%s.debug.log" % self.__dbName, "w" )
+      except IOError as e:
+        self.log.error( e )
+      
+    self.__client = Elasticsearch( self.__url, timeout = self.__timeout )
+    self.__tryToConnect()
+  
+  ########################################################################
+  def setClusterName( self, name ):
+    """
+      It is used to set the cluster name
+      
+      :param self: self reference
+      :param str requestName: request name
+    
+    """
+    self.__clusterName = name
+  
+  ########################################################################
+  def getClusterName( self ):
+    """
+    It returns the cluster name
+    
+    :param self: self reference
+    """
+    return self.__clusterName
+  
+  ########################################################################  
+  def query( self, index, query ):
+    """It exexutes a query and it returns the result
+    query is a dictionary. More info: search for elasticsearch dsl
+    
+    :param self: self reference
+    :param dict query: It is the query in ElasticSerach DSL language
+     
+    """
+    return self.__client.search( index = index, body = query )
+  
+  def _Search( self, indexname ):
+    """
+    it returns the object which can be used for reatriving ceratin value from the DB
+    """
+    return  Search( using = self.__client, index = indexname )
+  
+  ########################################################################
+  def _Q( self, name_or_query = 'match', **params ):
+    """
+    It is a wrapper to ElasticDSL Query module used to create a query object. 
+    :param name_or_query str is the type of the query
```

Comment:
The parameter description needs a colon before the description text. also the type goes after param before the name: i.e.:
```
:param str name_or_query: the type of the query
```
Also newline before the param thing. There are other docstrings where the second colon is missing